### PR TITLE
Replace vec2, vec3, and xColor properties with JS prototypes

### PR DESCRIPTION
**Pull request by SamGondelman**
Fixes:
https://highfidelity.manuscript.com/f/cases/16341/vec3-saving-with-blue-green-red-info-in-json
https://highfidelity.manuscript.com/f/cases/10596/Entities-editEntity-color-changes-should-support-r-g-b-in-addition-to-spelling-out-red-green-blue
https://highfidelity.manuscript.com/f/cases/12274/PolyLine-entity-s-strokeColors-property-should-be-an-array-of-color-values-not-Vec3-values
https://highfidelity.manuscript.com/f/cases/16358/Entity-color-edits-not-sticking
https://highfidelity.manuscript.com/f/cases/17073/MyAvatar-position-returns-an-object-with-keys-red-green-blue-and-x-y-z
https://highfidelity.manuscript.com/f/cases/17003/Export-Entity-JSON-file-exports-both-xyz-AND-rgb

Before, our JS vec3s were just maps with x, y, and z properties.  There was also a color type with only red, green, and blue.  Now, vec2s and vec3s are exposed to JS as objects with properties (x, y, z) and aliases (r, g, b)/(red, green, blue).  Colors will stringify with red/green/blue and other vectors will stringify with x/y/z.  Similarly, vec2s have (x, y) and alias (u, v).

What does this mean?

Before:
```
var color = Entities.getEntityProperties(<entity id>).color;
print(JSON.stringify(color)); // prints `{red: 255, green: 0, blue: 0}`
color.r = 127;
print(JSON.stringify(color)); // prints `{red: 255, green: 0, blue: 0, r: 127}`
Entities.editEntity(<entity id>, { color: {red: 0, green: 255, blue: 0} }); // works
Entities.editEntity(<entity id>, { color: {r: 0, g: 255, b: 0} }); // doesn't work
Entities.editEntity(<entity id>, { color: {x: 0, y: 255, z: 0} }); // doesn't work
Entities.editEntity(<entity id>, { color: [0, 255, 0]); // doesn't work
```
Now:
```
var color = Entities.getEntityProperties(<entity id>).color;
print(JSON.stringify(color)); // prints `{red: 255, green: 0, blue: 0}`
color.r = 127;
print(JSON.stringify(color)); // prints `{red: 127, green: 0, blue: 0}`
Entities.editEntity(<entity id>, { color: {red: 0, green: 255, blue: 0} }); // works
Entities.editEntity(<entity id>, { color: {r: 0, g: 255, b: 0} }); // works
Entities.editEntity(<entity id>, { color: {x: 0, y: 255, z: 0} }); // works
Entities.editEntity(<entity id>, { color: [0, 255, 0]); // works
```

Test plan:
- These changes affect a lot of different areas.  Run the autotester to confirm that the tests in hifi_tests still work.
- Verify that Create still works properly, with special attention to vec2 or vec3 properties: position, dimensions, various physics properties, and particle properties (verify that the particle tab works correctly, especially the color fields and color pickers)
- Export some entities (try: primitives and particles after messing with their properties).  In the resulting JSON, confirm that vec3 properties only list x, y, and z (not both xyz and red/green/blue).  Colors should list red/green/blue (check: cube color, particle color/colorSpread/colorStart/colorFinish, text textColor/backgroundColor, zone skybox color/haze color/keylight color).  Import those entities again and confirm that the properties persisted correctly.
- Confirm that you can edit an entity's color like this: `Entities.editEntity(<entity id>, { color: { x: 255, green: 155, b: 55 } });`.  Go into create and confirm that its color is `{ red: 255, green: 155, blue: 55 }`
- Create a sphere.  Sphere is red. Shut down Interface and server.  Restart server and Interface.  Edit sphere's color: change green component to 255, then change blue component to 255.  Sphere is white. 
 Restart server and Interface.  Sphere should be white (not red, as in master).
- In the console, enter `JSON.stringify(MyAvatar.position)`.  The result should only have xyz.
- Verify that Polylines still work.  Try the Fingerpaint app.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11285/HighFidelity-Beta-PR13596-43c71d5.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11285/HighFidelity-Beta-PR13596-43c71d5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11285/HighFidelity-Beta-Interface-PR13596-43c71d5.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11290/HighFidelity-Beta-PR13596-63fd9e8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11290/HighFidelity-Beta-Interface-PR13596-63fd9e8.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11295/HighFidelity-Beta-PR13596-353ff92.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11295/HighFidelity-Beta-PR13596-353ff92.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11295/HighFidelity-Beta-Interface-PR13596-353ff92.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11295/HighFidelity-Beta-PR13596-353ff92.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11295/HighFidelity-Beta-Interface-PR13596-353ff92.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11296/HighFidelity-Beta-PR13596-f9cdfba.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11296/HighFidelity-Beta-PR13596-f9cdfba.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11296/HighFidelity-Beta-Interface-PR13596-f9cdfba.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11296/HighFidelity-Beta-PR13596-f9cdfba.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11296/HighFidelity-Beta-Interface-PR13596-f9cdfba.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11718/HighFidelity-Beta-PR13596-3768714.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11721/HighFidelity-Beta-PR13596-cde9dd3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11721/HighFidelity-Beta-PR13596-cde9dd3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11721/HighFidelity-Beta-Interface-PR13596-cde9dd3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11721/HighFidelity-Beta-PR13596-cde9dd3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11721/HighFidelity-Beta-Interface-PR13596-cde9dd3.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11740/HighFidelity-Beta-PR13596-7c8d080.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11740/HighFidelity-Beta-Interface-PR13596-7c8d080.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11740/HighFidelity-Beta-PR13596-7c8d080.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11740/HighFidelity-Beta-Interface-PR13596-7c8d080.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11842/HighFidelity-Beta-PR13596-7772e8a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11842/HighFidelity-Beta-PR13596-7772e8a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11842/HighFidelity-Beta-Interface-PR13596-7772e8a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11842/HighFidelity-Beta-PR13596-7772e8a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11842/HighFidelity-Beta-Interface-PR13596-7772e8a.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Test plan passes

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11882/HighFidelity-Beta-PR13596-9427de1.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11882/HighFidelity-Beta-PR13596-9427de1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11882/HighFidelity-Beta-Interface-PR13596-9427de1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11882/HighFidelity-Beta-PR13596-9427de1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11882/HighFidelity-Beta-Interface-PR13596-9427de1.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by conklin94122** [comment]
Removing QA Approved, will need to be retested in master/73.0

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12337/HighFidelity-Beta-PR13596-07e5edf.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12337/HighFidelity-Beta-PR13596-07e5edf.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12337/HighFidelity-Beta-Interface-PR13596-07e5edf.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12337/HighFidelity-Beta-PR13596-07e5edf.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12337/HighFidelity-Beta-Interface-PR13596-07e5edf.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12349/HighFidelity-Beta-PR13596-283c0c0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12349/HighFidelity-Beta-PR13596-283c0c0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12349/HighFidelity-Beta-Interface-PR13596-283c0c0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12349/HighFidelity-Beta-PR13596-283c0c0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12349/HighFidelity-Beta-Interface-PR13596-283c0c0.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12416/HighFidelity-Beta-PR13596-6730635.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12416/HighFidelity-Beta-PR13596-6730635.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12416/HighFidelity-Beta-Interface-PR13596-6730635.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12416/HighFidelity-Beta-PR13596-6730635.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12416/HighFidelity-Beta-Interface-PR13596-6730635.exe

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by qaAdamSmith82** [comment]
Releasing Testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12648/HighFidelity-Beta-PR13596-9b75c7e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12648/HighFidelity-Beta-PR13596-9b75c7e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12648/HighFidelity-Beta-Interface-PR13596-9b75c7e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12652/HighFidelity-Beta-PR13596-589e88a.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12648/HighFidelity-Beta-PR13596-9b75c7e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12648/HighFidelity-Beta-Interface-PR13596-9b75c7e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12652/HighFidelity-Beta-PR13596-589e88a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12652/HighFidelity-Beta-Interface-PR13596-589e88a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12652/HighFidelity-Beta-PR13596-589e88a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12652/HighFidelity-Beta-Interface-PR13596-589e88a.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12672/HighFidelity-Beta-PR13596-15162dc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12672/HighFidelity-Beta-PR13596-15162dc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12672/HighFidelity-Beta-Interface-PR13596-15162dc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12672/HighFidelity-Beta-PR13596-15162dc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12672/HighFidelity-Beta-Interface-PR13596-15162dc.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12794/HighFidelity-Beta-PR13596-1c74a5e.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12795/HighFidelity-Beta-PR13596-b3bdddb.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12795/HighFidelity-Beta-PR13596-b3bdddb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12795/HighFidelity-Beta-Interface-PR13596-b3bdddb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12795/HighFidelity-Beta-PR13596-b3bdddb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12795/HighFidelity-Beta-Interface-PR13596-b3bdddb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12997/HighFidelity-Beta-PR13596-c522d1b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12997/HighFidelity-Beta-Interface-PR13596-c522d1b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12997/HighFidelity-Beta-PR13596-c522d1b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12997/HighFidelity-Beta-Interface-PR13596-c522d1b.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13000/HighFidelity-Beta-PR13596-c522d1b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13000/HighFidelity-Beta-Interface-PR13596-c522d1b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13000/HighFidelity-Beta-PR13596-c522d1b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13000/HighFidelity-Beta-Interface-PR13596-c522d1b.exe

**Comment by samcake** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13048/HighFidelity-Beta-PR13596-c522d1b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13048/HighFidelity-Beta-PR13596-c522d1b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13048/HighFidelity-Beta-Interface-PR13596-c522d1b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13048/HighFidelity-Beta-PR13596-c522d1b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13048/HighFidelity-Beta-Interface-PR13596-c522d1b.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13559/HighFidelity-Beta-PR13596-5b6e7b2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13559/HighFidelity-Beta-PR13596-5b6e7b2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13559/HighFidelity-Beta-Interface-PR13596-5b6e7b2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13559/HighFidelity-Beta-PR13596-5b6e7b2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13559/HighFidelity-Beta-Interface-PR13596-5b6e7b2.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13630/HighFidelity-Beta-PR13596-4e96c32.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13630/HighFidelity-Beta-PR13596-4e96c32.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13630/HighFidelity-Beta-Interface-PR13596-4e96c32.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13630/HighFidelity-Beta-PR13596-4e96c32.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13630/HighFidelity-Beta-Interface-PR13596-4e96c32.exe

**Comment by huffman** [comment]
Here are my initial concerns and questions after looking over the PR:

 1. I still don't fully understand why we can't use vectors internally but different types in the JS API? I would we could handle the exceptions where needed, like with the particle properties.
 1. I personally think that having extra flexibility in how you can specify and access colors/vec3s/etc. is a net negative. I briefly commented on this in the #developers channel. I think Seth prefers it, but I think that in the long run it means:
    1. When looking at sample code, there might be a discrepancy in how users specify or read these values ([1, 2] vs {x: 1, y: 2}). This can be unnecessarily confusing for new users.
    1. Our C++ <-> JS marshalling is where we spend a large portion of our time executing in scripts, and this is adding more overhead for one of our core types. Once we add it, it will be hard to remove it.
    1. Sometimes you will be able to use the special accessors, and sometimes you won't. It won't always be obvious to new users when that is.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13708/HighFidelity-Beta-PR13596-67662df.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13708/HighFidelity-Beta-PR13596-67662df.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13708/HighFidelity-Beta-Interface-PR13596-67662df.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13708/HighFidelity-Beta-PR13596-67662df.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13708/HighFidelity-Beta-Interface-PR13596-67662df.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13791/HighFidelity-Beta-PR13596-9788aa3.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13791/HighFidelity-Beta-PR13596-9788aa3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13791/HighFidelity-Beta-Interface-PR13596-9788aa3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13791/HighFidelity-Beta-PR13596-9788aa3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13791/HighFidelity-Beta-Interface-PR13596-9788aa3.exe

File: libraries/entities/src/EntityItemProperties.cpp [hunk 1]
```diff
@@ -1314,9 +1314,6 @@ QScriptValue EntityItemProperties::copyToScriptValue(QScriptEngine* engine, bool
     COPY_PROPERTY_TO_QSCRIPTVALUE(PROP_NAME, name);
     COPY_PROPERTY_TO_QSCRIPTVALUE(PROP_COLLISION_SOUND_URL, collisionSoundURL);
 
-    // Light, Line, Model, ParticleEffect, PolyLine, Shape
-    COPY_PROPERTY_TO_QSCRIPTVALUE(PROP_COLOR, color);
```
Comment (huffman) [inline libraries/entities/src/EntityItemProperties.cpp]: While this value may not have meaning for some types, I'm concerned that we are changing the API to not include colors for certain types of entities. Does this affect entity properties at all, for entities not in this list?

Comment (SamGondelman) [reply libraries/entities/src/EntityItemProperties.cpp]: If I understand your concern correctly, no.  In master, if you do `Entities.getEntityProperties("<id of web entity>").color` you'd get `undefined`.  It's the same in this PR.

Comment (huffman) [reply libraries/entities/src/EntityItemProperties.cpp]: Ok, I did not know that. Sounds good to me then.

File: interface/src/raypick/ParabolaPointer.cpp [hunk 2]
```diff
@@ -53,7 +53,7 @@ void ParabolaPointer::editRenderStatePath(const std::string& state, const QVaria
             enabled = true;
             if (pathMap["color"].isValid()) {
                 bool valid;
-                color = toGlm(xColorFromVariant(pathMap["color"], valid));
+                color = toGlm(vec3FromVariant(pathMap["color"], valid));
```
Comment (jherico) [inline interface/src/raypick/ParabolaPointer.cpp]: i don't understand why there still has to be a `toGlm` call here.

File: interface/src/scripting/SelectionScriptingInterface.cpp [hunk 3]
```diff
@@ -497,10 +497,11 @@ bool SelectionHighlightStyle::fromVariantMap(const QVariantMap& properties) {
 QVariantMap SelectionHighlightStyle::toVariantMap() const {
     QVariantMap properties;
 
-    properties["outlineUnoccludedColor"] = xColorToVariant(xColorFromGlm(_style._outlineUnoccluded.color));
-    properties["outlineOccludedColor"] = xColorToVariant(xColorFromGlm(_style._outlineOccluded.color));
-    properties["fillUnoccludedColor"] = xColorToVariant(xColorFromGlm(_style._fillUnoccluded.color));
-    properties["fillOccludedColor"] = xColorToVariant(xColorFromGlm(_style._fillOccluded.color));
+    const float MAX_COLOR = 255.0f;
+    properties["outlineUnoccludedColor"] = vec3toVariant(_style._outlineUnoccluded.color * MAX_COLOR);
```
Comment (jherico) [inline interface/src/scripting/SelectionScriptingInterface.cpp]: can we not overload vec3toVariant to distinguish between passing a `glm::vec3` and a `glm::u8vec3`, so we don't have to do this MAX_COLOR multiply?

Comment (SamGondelman) [reply interface/src/scripting/SelectionScriptingInterface.cpp]: no, that would be an ambiguous overload

File: interface/src/raypick/ParabolaPointer.cpp (hunk 2 above)
Comment (SamGondelman) [reply interface/src/raypick/ParabolaPointer.cpp]: toGlm actually divides by 255, but it's doing an extra conversion here.  I'll fix it.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13964/HighFidelity-Beta-PR13596-ded8d4b.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13964/HighFidelity-Beta-PR13596-ded8d4b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13964/HighFidelity-Beta-Interface-PR13596-ded8d4b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13964/HighFidelity-Beta-PR13596-ded8d4b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13964/HighFidelity-Beta-Interface-PR13596-ded8d4b.dmg

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14028/HighFidelity-Beta-PR13596-e3543cf.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14028/HighFidelity-Beta-Interface-PR13596-e3543cf.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14028/HighFidelity-Beta-PR13596-e3543cf.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14028/HighFidelity-Beta-Interface-PR13596-e3543cf.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14033/HighFidelity-Beta-PR13596-3f117cd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14033/HighFidelity-Beta-Interface-PR13596-3f117cd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14033/HighFidelity-Beta-PR13596-3f117cd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14033/HighFidelity-Beta-Interface-PR13596-3f117cd.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14043/HighFidelity-Beta-PR13596-3f117cd.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14043/HighFidelity-Beta-PR13596-3f117cd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14043/HighFidelity-Beta-Interface-PR13596-3f117cd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14043/HighFidelity-Beta-PR13596-3f117cd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14043/HighFidelity-Beta-Interface-PR13596-3f117cd.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Test plan passes. Auto Tests were run and passed by @qaAdamSmith82

**Pull request merged**